cmd/mattermost/commands/webhook.go

```diff
@@ -75,9 +84,48 @@ func listWebhookCmdF(command *cobra.Command, args []string) error {
 	return nil
 }
 
+func createIncomingWebhookCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer app.Shutdown()
+
+	channelArg, _ := command.Flags().GetString("channel")
+	channel := getChannelFromChannelArg(app, channelArg)
+	if channel == nil {
+		return errors.New("Unable to find channel '" + channelArg + "'")
+	}
+
+	userArg, _ := command.Flags().GetString("user")
+	description, _ := command.Flags().GetString("description")
+	iconUrl, _ := command.Flags().GetString("icon")
+	channelLocked, _ := command.Flags().GetBool("lock-to-channel")
+
+	incomingWebhook := &model.IncomingWebhook{
+		ChannelId:     channel.Id,
+		Description:   description,
+		IconURL:       iconUrl,
+		ChannelLocked: channelLocked,
+	}
+
+	if _, err := app.CreateIncomingWebhookForChannel(userArg, channel, incomingWebhook); err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func init() {
+	WebhookCreateIncomingCmd.Flags().String("channel", "", "Channel ID")
+	WebhookCreateIncomingCmd.Flags().String("user", "", "User ID")
+	WebhookCreateIncomingCmd.Flags().String("description", "", "Incoming webhook description")
+	WebhookCreateIncomingCmd.Flags().String("icon", "", "Icon URL")
+	WebhookCreateIncomingCmd.Flags().Bool("lock-to-channel", false, "Lock to channel")
```

Comment:
Please could you also add a parameter for display name, as this is what will show up for the webhook in the web UI.

Reply:
Added display name parameter